opencti-platform/opencti-graphql/src/database/engine.js

```diff
@@ -1965,43 +1977,76 @@ const completeSpecialFilterKeys = async (context, user, inputFilters) => {
+      if (arrayKeys.includes(IDS_FILTER)) {
+        // the special filter key 'ids' take all the ids into account
+        const { newFilter, newFilterGroup } = adaptFilterToIdsFilterKey(filter);
+        if (newFilter) {
+          finalFilters.push(newFilter);
+        }
+        if (newFilterGroup) {
+          finalFilterGroups.push(newFilterGroup);
+        }
       }
-    } else if (arrayKeys.includes(IDS_FILTER)) {
-      // the special filter key 'ids' take all the ids into account
-      const { newFilter, newFilterGroup } = adaptFilterToIdsFilterKey(filter);
-      if (newFilter) {
+      if (arrayKeys.includes(TYPE_FILTER)) {
+        // in case we want to filter by entity_type
+        // we need to add parent_types checking (in case the given value in type is an abstract type)
+        const { newFilter, newFilterGroup } = adaptFilterToEntityTypeFilterKey(filter);
+        if (newFilter) {
+          finalFilters.push(newFilter);
+        }
+        if (newFilterGroup) {
+          finalFilterGroups.push(newFilterGroup);
+        }
+      }
+      if (arrayKeys.includes(WORKFLOW_FILTER) || arrayKeys.includes(X_OPENCTI_WORKFLOW_ID)) {
+        // in case we want to filter by status template (template of a workflow status) or status
+        // we need to find all statuses filtered by status template and filter on these statuses
+        const newFilter = await adaptFilterToWorkflowFilterKey(context, user, filter);
         finalFilters.push(newFilter);
       }
-      if (newFilterGroup) {
-        finalFilterGroups.push(newFilterGroup);
+      if (arrayKeys.includes(SOURCE_RELIABILITY_FILTER)) {
+        // in case we want to filter by source reliability (reliability of author)
+        // we need to find all authors filtered by reliability and filter on these authors
+        const { newFilter, newFilterGroup } = await adaptFilterToSourceReliabilityFilterKey(context, user, filter);
+        if (newFilter) {
+          finalFilters.push(newFilter);
+        }
+        if (newFilterGroup) {
+          finalFilterGroups.push(newFilterGroup);
+        }
       }
-    } else if (arrayKeys.includes(SOURCE_RELIABILITY_FILTER)) {
-      // in case we want to filter by source reliability (reliability of author)
-      // we need to find all authors filtered by reliability and filter on these authors
-      const { newFilter, newFilterGroup } = await adaptFilterToSourceReliabilityFilterKey(context, user, filter);
-      if (newFilter) {
-        finalFilters.push(newFilter);
+      if (arrayKeys.includes(INSTANCE_FILTER)) {
+        const nested = [{ key: 'internal_id', operator: filter.operator, values: filter.values }];
+        finalFilters.push({ key: 'connections', nested });
+      }
+      if (arrayKeys.includes(RELATION_FROM_FILTER) || arrayKeys.includes(RELATION_TO_FILTER)
+          || arrayKeys.includes(RELATION_TO_SIGHTING_FILTER)) {
+        const side = arrayKeys.includes(RELATION_FROM_FILTER) ? 'from' : 'to';
```

Comment:
if there is both a filter on fromId and on toId, only the from will be taken into account